### PR TITLE
smokeb2c guest user scenarios

### PR DESCRIPTION
**Pull request by khushal-rathore-metacube**
https://spryker.atlassian.net/browse/CC-21837

Covered below smoke b2c guest user scenarios:

1. Add to cart products as a guest user and place order as a guest user
2. Add to cart products as a guest user and login during checkout
3. Add to cart products as a guest user and register during checkout
4. Check guest has no access to wishlist → click on button redirects to login page

### CODE REVIEW

File: tests/smoke/smoke_b2c.robot [hunk 1]
```diff
@@ -598,4 +598,69 @@ Refunds
     Zed: grand total for the order equals:    ${lastPlacedOrder}    €0.00
     [Teardown]    Run keywords    Zed: login on Zed with provided credentials:    ${zed_admin_email}
     ...    AND    Zed: activate following discounts from Overview page:    Tu & Wed $5 off 5 or more    10% off $100+    20% off cameras    Tu & Wed €5 off 5 or more    10% off minimum order
- 
+ Add_to_cart_products_as_a_guest_user_and_place_order_as_a_guest_user
```
Comment (OKrivtsova) [inline tests/smoke/smoke_b2c.robot:601]: we already have a test for guest checkout - 'Guest_Checkout'
you are not supposed to cover things which are already covered

File: tests/smoke/smoke_b2c.robot [hunk 2]
```diff
@@ -598,4 +598,69 @@ Refunds
...
+    Yves: submit form on the checkout
+    Yves: select the following shipping method on the checkout and go next:    Express
+    Yves: select the following payment method on the checkout and go next:    Invoice
+    Yves: accept the terms and conditions:    true
+    Yves: 'submit the order' on the summary page
+    Yves: 'Thank you' page is displayed
+     [Teardown]    Zed: delete customer:
+    ...    || email                          ||
+    ...    || abc${random}@gmail.com ||
+Check_guest_has_no_access_to_wishlist
```
Comment (OKrivtsova) [inline tests/smoke/smoke_b2c.robot:661]: please add this check into one of the previous tests (there you can check it before you start checkout) we do not need it a s a separate test

File: resources/steps/checkout_steps.robot [hunk 3]
```diff
@@ -203,6 +203,25 @@ Yves: assert merchant of product in cart or list:
...
+    Click Element by xpath with JavaScript    //span[normalize-space()='Sign Up']
+    Type Text    ${yves_checkout_signup_first_name}    ${firstname}
+    Type Text    ${yves_checkout_signup_last_name}    ${lastname}
+    Type Text    ${yves_checkout_signup_email}    ${email}
+    Type Text    ${yves_checkout_signup_password}    ${password}
+    Type Text    ${yves_checkout_signup_confirm_password}    ${confirmpassword}
+    Wait Until Element Is Visible   ${yves_checkout_signup_accept_terms}
+    Check Checkbox  ${yves_checkout_signup_accept_terms}
+    Click    ${yves_checkout_signup_tab}   
+Yves: click on button redirects to login page
```
[the remark this answers is not in the document]
Comment (khushal-rathore-metacube) [reply resources/steps/checkout_steps.robot:225]: Done

File: resources/pages/yves/yves_checkout_login_page.robot [hunk 4]
```diff
@@ -2,4 +2,13 @@
...
+${yves_checkout_login_buy_as_guest_submit_button}    xpath=//button[@type='submit' and contains(text(),'Buy as Guest')]
+${yves_checkout_signup_button}    xpath=//span[normalize-space()='Sign Up']
+${yves_checkout_signup_first_name}    id=registerForm_customer_first_name
+${yves_checkout_signup_last_name}    id=registerForm_customer_last_name
+${yves_checkout_signup_email}        id=registerForm_customer_email
+${yves_checkout_signup_password}    id=registerForm_customer_password_pass
+${yves_checkout_signup_confirm_password}    id=registerForm_customer_password_confirm
+${yves_checkout_signup_accept_terms}    xpath=(//span[@class='checkbox__box'])[1]
+${yves_checkout_signup_tab}    xpath=//button[normalize-space()='Sign Up']
+${yves_checkout_login_tab}    xpath=//span[normalize-space()='Login']
```
[the remark this answers is not in the document]
Comment (khushal-rathore-metacube) [reply resources/pages/yves/yves_checkout_login_page.robot:14]: Done

File: resources/steps/checkout_steps.robot [hunk 5]
```diff
@@ -203,6 +203,25 @@ Yves: assert merchant of product in cart or list:
...
-
-
+ Yves: proceed with checkout and login as a guest user:
+    [Arguments]    ${email}    ${password}=${default_password}   
+    Wait Until Page Contains Element    ${yves_checkout_login_tab} 
+    Click Element by xpath with JavaScript    //span[normalize-space()='Login']
+    Type Text    ${email_field}    ${email}
+    Type Text    ${password_field}    ${password}
+    Click    ${form_login_button}
+Yves: proceed with checkout as guest user:
```
[the remark this answers is not in the document]
Comment (khushal-rathore-metacube) [reply resources/steps/checkout_steps.robot:213]: Done

File: tests/smoke/smoke_b2c.robot (hunk 1 above)
Comment (khushal-rathore-metacube) [reply tests/smoke/smoke_b2c.robot:601]: Done

File: resources/common/common_yves.robot [hunk 6]
```diff
@@ -25,8 +26,15 @@ Resource    ../steps/header_steps.robot
...
-
+${default_db_host}         127.0.0.1
+${default_db_name}         eu-docker
+${default_db_password}     secret
+${default_db_port}         3306
+${default_db_user}         spryker
+${api_timeout}             60
+${default_allow_redirects}     true
+${default_auth}                ${NONE}
+${current_url}        http://glue.de.spryker.local
```
[the remark this answers is not in the document]
Comment (khushal-rathore-metacube) [reply resources/common/common_yves.robot:37]: Done, keeping as it is , as updating it leads to conflict in setup

File: resources/common/common_yves.robot [hunk 7]
```diff
@@ -298,3 +306,53 @@ Yves: page should contain script with attribute:
...
+    ${var_value} =    Query    ${sql_query}
+    Disconnect From Database
+    ${var_value}=    Convert To String    ${var_value}
+    ${var_value}=    Replace String    ${var_value}    '   ${EMPTY}
+    ${var_value}=    Replace String    ${var_value}    ,   ${EMPTY}
+    ${var_value}=    Replace String    ${var_value}    (   ${EMPTY}
+    ${var_value}=    Replace String    ${var_value}    )   ${EMPTY}
+    Set Test Variable    ${${variable_name}}    ${var_value}
+    [Return]    ${variable_name}
+I send a POST request:
```
[the remark this answers is not in the document]
Comment (khushal-rathore-metacube) [reply resources/common/common_yves.robot:336]: Done, keeping as it is , as updating it leads to conflict in setup

File: resources/common/common_yves.robot [hunk 8]
```diff
@@ -298,3 +306,53 @@ Yves: page should contain script with attribute:
 Yves: page should contain script with id:
     [Arguments]    ${scriptId}
     Yves: page should contain script with attribute:    id    ${scriptId}
+Yves: login after signup during checkout:
+    [Arguments]    ${email}    ${password}
+    Type Text    ${email_field}     ${email}
+    Type Text    ${password_field}     ${password}
+    Click    ${form_login_button}
+Save the result of a SELECT DB query to a variable:
```
[the remark this answers is not in the document]
Comment (khushal-rathore-metacube) [reply resources/common/common_yves.robot:314]: keeping as it is , as updating it leads to conflict in setup

File: resources/steps/checkout_steps.robot [hunk 9]
```diff
@@ -203,6 +203,25 @@ Yves: assert merchant of product in cart or list:
     [Documentation]    Method for MP which asserts value in 'Sold by' label of item in cart or list. Requires concrete SKU
     [Arguments]    ${sku}    ${merchant_name_expected}
     Page Should Contain Element    xpath=//span[@itemprop='sku'][text()='${sku}']/../../following-sibling::p/a[text()='${merchant_name_expected}']
-
-
-
+ Yves: proceed with checkout and login as a guest user:
```
[the remark this answers is not in the document]
Comment (khushal-rathore-metacube) [reply resources/steps/checkout_steps.robot]: First did checkout and then login, renamed it

File: resources/steps/pdp_steps.robot [hunk 10]
```diff
@@ -177,4 +177,5 @@ Yves: merchant is (not) displaying in Sold By section of PDP:
     Wait Until Element Is Visible    ${pdp_product_sku}[${env}]
     Try reloading page until element is/not appear:    xpath=//section[@data-qa='component product-configurator']//div[contains(text(),'${merchantName}')]     ${condition}    20    5s  
    
-    
\ No newline at end of file
+Yves: add a product to cart on clicking add to cart button on PDP
```
[the remark this answers is not in the document]
Comment (khushal-rathore-metacube) [reply resources/steps/pdp_steps.robot:180]: This is not functioning on my scenario as the product does not have variant

File: tests/smoke/smoke_b2c.robot (hunk 2 above)
Comment (khushal-rathore-metacube) [reply tests/smoke/smoke_b2c.robot]: done

File: resources/pages/yves/yves_checkout_login_page.robot [hunk 11]
```diff
@@ -2,4 +2,13 @@
...
+${yves_checkout_login_buy_as_guest_submit_button}    xpath=//button[@type='submit' and contains(text(),'Buy as Guest')]
+${yves_checkout_signup_button}    xpath=//toggler-radio[@data-qa='component toggler-radio checkoutProceedAs register']//label[@class='toggler-radio__container']
+${yves_checkout_signup_first_name}    id=registerForm_customer_first_name
+${yves_checkout_signup_last_name}    id=registerForm_customer_last_name
+${yves_checkout_signup_email}        id=registerForm_customer_email
+${yves_checkout_signup_password}    id=registerForm_customer_password_pass
+${yves_checkout_signup_confirm_password}    id=registerForm_customer_password_confirm
+${yves_checkout_signup_accept_terms}    xpath=(//span[@class='checkbox__box'])[1]
+${yves_checkout_signup_tab}    xpath=//form[contains(@name,'registerForm')]//button[contains(@type,'submit')]
+${yves_checkout_login_tab}    xpath=//toggler-radio[@data-qa='component toggler-radio checkoutProceedAs login']//label[@class='toggler-radio__container']
```
Comment (OKrivtsova) [inline resources/pages/yves/yves_checkout_login_page.robot:14]: do we really need such a long xpath here? 
we can use this xpath here which is more flexible and still gets what we need it to get:
//toggler-radio[contains(@data-qa,'checkoutProceedAs login')

we do not need the label

File: resources/pages/yves/yves_checkout_login_page.robot [hunk 12]
```diff
@@ -2,4 +2,13 @@
 ${yves_checkout_login_guest_firstName_field}    id=guestForm_customer_first_name
 ${yves_checkout_login_guest_lastName_field}    id=guestForm_customer_last_name
 ${yves_checkout_login_guest_email_field}    id=guestForm_customer_email
-${yves_checkout_login_buy_as_guest_submit_button}    xpath=//button[@type='submit' and contains(text(),'Buy as Guest')]
\ No newline at end of file
+${yves_checkout_login_buy_as_guest_submit_button}    xpath=//button[@type='submit' and contains(text(),'Buy as Guest')]
+${yves_checkout_signup_button}    xpath=//toggler-radio[@data-qa='component toggler-radio checkoutProceedAs register']//label[@class='toggler-radio__container']
```
Comment (OKrivtsova) [inline resources/pages/yves/yves_checkout_login_page.robot:6]: we do not need such an xpath here
//toggler-radio[contains(@data-qa,'checkoutProceedAs register') - will do

File: resources/steps/checkout_steps.robot [hunk 13]
```diff
@@ -203,6 +205,25 @@ Yves: assert merchant of product in cart or list:
     [Documentation]    Method for MP which asserts value in 'Sold by' label of item in cart or list. Requires concrete SKU
     [Arguments]    ${sku}    ${merchant_name_expected}
     Page Should Contain Element    xpath=//span[@itemprop='sku'][text()='${sku}']/../../following-sibling::p/a[text()='${merchant_name_expected}']
-
-
-
+ Yves: proceed as a guest user and login during checkout:
```
Comment (OKrivtsova) [inline resources/steps/checkout_steps.robot:208]: there needs to be an empty line between tests for readability

File: resources/steps/checkout_steps.robot [hunk 14]
```diff
@@ -203,6 +205,25 @@ Yves: assert merchant of product in cart or list:
...
+    [Arguments]    ${email}    ${password}=${default_password}   
+    Wait Until Page Contains Element    ${yves_checkout_login_tab} 
+    Click Element by xpath with JavaScript    //span[normalize-space()='Login']
```
Comment (OKrivtsova) [inline resources/steps/checkout_steps.robot:211]: - the xpath should be defined in the page object
- the xpath should not depend on a label if we can avoid it

File: resources/steps/checkout_steps.robot [hunk 15]
```diff
@@ -203,6 +205,25 @@ Yves: assert merchant of product in cart or list:
...
+    Type Text    ${email_field}    ${email}
+    Type Text    ${password_field}    ${password}
+    Click    ${form_login_button}
+Yves: signup guest user during checkout:
+      [Arguments]      ${firstName}    ${lastName}    ${email}     ${password}      ${confirmpassword} 
+    Wait Until Page Contains Element    ${yves_checkout_signup_button} 
+    Click Element by xpath with JavaScript    //span[normalize-space()='Sign Up']
```
Comment (OKrivtsova) [inline resources/steps/checkout_steps.robot:218]: the xpath should be defined in the page object
the xpath should not depend on a label if we can avoid it

File: tests/smoke/smoke_b2c.robot [hunk 16]
```diff
@@ -598,4 +599,50 @@ Refunds
...
+Add_to_cart_products_as_a_guest_user_and_login_during_checkout
+    Yves: go to the 'Home' page
+    Yves: go to PDP of the product with sku:    207_15721464
+    Yves: add a product to cart on clicking add to cart button on PDP
+    Yves: go to b2c shopping cart  
+    Yves: click on the 'Checkout' button in the shopping cart
+     Yves: proceed as a guest user and login during checkout:   ${yves_second_user_email}
+     Yves: fill in the following new shipping address:
+    ...    || salutation | firstName | lastName | street        | houseNumber | postCode | city   | country | company | phone     | additionalAddress ||
+    ...    || Mr.        | Guest     | User     | Kirncher Str. | 7           | 10247    | Berlin | Germany | Spryker | 123456789 | Additional street ||
```
Comment (OKrivtsova) [inline tests/smoke/smoke_b2c.robot:612]: why are all the address fields hardcoded? we have variables for that

File: tests/smoke/smoke_b2c.robot [hunk 17]
```diff
@@ -598,4 +599,50 @@ Refunds
     Zed: grand total for the order equals:    ${lastPlacedOrder}    €0.00
     [Teardown]    Run keywords    Zed: login on Zed with provided credentials:    ${zed_admin_email}
     ...    AND    Zed: activate following discounts from Overview page:    Tu & Wed $5 off 5 or more    10% off $100+    20% off cameras    Tu & Wed €5 off 5 or more    10% off minimum order
- 
+
+Add_to_cart_products_as_a_guest_user_and_login_during_checkout
+    Yves: go to the 'Home' page
+    Yves: go to PDP of the product with sku:    207_15721464
```
Comment (OKrivtsova) [inline tests/smoke/smoke_b2c.robot:605]: sku should not be hardcoded, we have variables for that

File: tests/smoke/smoke_b2c.robot [hunk 18]
```diff
@@ -598,4 +599,50 @@ Refunds
...
+     Yves: fill in the following new shipping address:
+    ...    || salutation | firstName | lastName | street        | houseNumber | postCode | city   | country | company | phone     | additionalAddress ||
+    ...    || Mr.        | Guest     | User     | Kirncher Str. | 7           | 10247    | Berlin | Germany | Spryker | 123456789 | Additional street ||
+    Yves: submit form on the checkout
+    Yves: select the following shipping method on the checkout and go next:    Express
+    Yves: select the following payment method on the checkout and go next:    Invoice
+    Yves: accept the terms and conditions:    true
+    Yves: 'submit the order' on the summary page
+    Yves: 'Thank you' page is displayed
+    [Teardown]    Run Keywords    Delete All Cookies
```
Comment (OKrivtsova) [inline tests/smoke/smoke_b2c.robot:619]: why do we need this teardown?

File: tests/smoke/smoke_b2c.robot [hunk 19]
```diff
@@ -598,4 +599,50 @@ Refunds
...
+    Page Should Not Contain Element    ${pdp_add_to_wishlist_button}
+    Yves: go to b2c shopping cart  
+    Yves: click on the 'Checkout' button in the shopping cart
+    Yves: signup guest user during checkout:    test    user    abc${random}@gmail.com    Abc#${random}    Abc#${random}
+    Save the result of a SELECT DB query to a variable:    select registration_key from spy_customer where email = 'abc${random}@gmail.com'    confirmation_key
+    I send a POST request:     /customer-confirmation   {"data":{"type":"customer-confirmation","attributes":{"registrationKey":"${confirmation_key}"}}}
+    Yves: login after signup during checkout:    abc${random}@gmail.com    Abc#${random}
+    Yves: fill in the following new shipping address:
+    ...    || salutation | firstName | lastName | street        | houseNumber | postCode | city   | country | company | phone     | additionalAddress ||
+    ...    || Mr.        | Guest     | User     | Kirncher Str. | 7           | 10247    | Berlin | Germany | Spryker | 123456789 | Additional street ||
```
Comment (OKrivtsova) [inline tests/smoke/smoke_b2c.robot:634]: why are all the address fields hardcoded? we have variables for that

File: tests/smoke/smoke_b2c.robot [hunk 20]
```diff
@@ -598,4 +599,50 @@ Refunds
...
+    Yves: select the following shipping method on the checkout and go next:    Express
+    Yves: select the following payment method on the checkout and go next:    Invoice
+    Yves: accept the terms and conditions:    true
+    Yves: 'submit the order' on the summary page
+    Yves: 'Thank you' page is displayed
+    [Teardown]    Run Keywords    Delete All Cookies
+    ...    AND       Reload
+Add_to_cart_products_as_a_guest_user_and_register_during_checkout
+    Yves: go to the 'Home' page
+    Yves: go to PDP of the product with sku:    207_15721464
```
Comment (OKrivtsova) [inline tests/smoke/smoke_b2c.robot:623]: sku should be in a variable

File: tests/smoke/smoke_b2c.robot [hunk 21]
```diff
@@ -598,4 +599,50 @@ Refunds
...
+    ...    || Mr.        | Guest     | User     | Kirncher Str. | 7           | 10247    | Berlin | Germany | Spryker | 123456789 | Additional street ||
+    Yves: submit form on the checkout
+    Yves: select the following shipping method on the checkout and go next:    Express
+    Yves: select the following payment method on the checkout and go next:    Invoice
+    Yves: accept the terms and conditions:    true
+    Yves: 'submit the order' on the summary page
+    Yves: 'Thank you' page is displayed
+    Yves: go to the 'Home' page
+    Yves: login on Yves with provided credentials:    ${yves_second_user_email}
+    Yves: go to PDP of the product with sku:    207_15721464
```
Comment (OKrivtsova) [inline tests/smoke/smoke_b2c.robot:643]: sku should be a variable

Comment (khushal-rathore-metacube) [reply tests/smoke/smoke_b2c.robot:643]: Done

File: tests/smoke/smoke_b2c.robot (hunk 20 above)
Comment (khushal-rathore-metacube) [reply tests/smoke/smoke_b2c.robot:623]: Done

File: resources/steps/checkout_steps.robot (hunk 13 above)
Comment (khushal-rathore-metacube) [reply resources/steps/checkout_steps.robot:208]: Done

File: resources/steps/checkout_steps.robot (hunk 14 above)
Comment (khushal-rathore-metacube) [reply resources/steps/checkout_steps.robot:211]: Done

File: resources/steps/checkout_steps.robot (hunk 15 above)
Comment (khushal-rathore-metacube) [reply resources/steps/checkout_steps.robot:218]: Done

File: resources/pages/yves/yves_checkout_login_page.robot (hunk 12 above)
Comment (khushal-rathore-metacube) [reply resources/pages/yves/yves_checkout_login_page.robot:6]: Done

File: resources/pages/yves/yves_checkout_login_page.robot (hunk 11 above)
Comment (khushal-rathore-metacube) [reply resources/pages/yves/yves_checkout_login_page.robot:14]: Done

File: tests/smoke/smoke_b2c.robot (hunk 17 above)
Comment (khushal-rathore-metacube) [reply tests/smoke/smoke_b2c.robot:605]: Done

File: tests/smoke/smoke_b2c.robot (hunk 16 above)
Comment (khushal-rathore-metacube) [reply tests/smoke/smoke_b2c.robot:612]: Done

**Review by OKrivtsova — DISMISSED** [review]

**Comment by khushal-rathore-metacube** [comment]
Resolved conflicts

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**